sql/core/src/main/scala/org/apache/spark/sql/execution/command/DataWritingCommand.scala

```diff
@@ -96,4 +98,20 @@ object DataWritingCommand {
       sparkContext.getLocalProperty(SQLExecution.EXECUTION_ID_KEY),
       metrics.values.toSeq)
   }
+
+  def assertEmptyRootPath(locationUri: Option[URI], configuration: Configuration) {
+    locationUri.fold() { tablePathUri: URI =>
+      val tablePath = tablePathUri.toString
+      val filePath = new org.apache.hadoop.fs.Path(tablePath)
+      val fs = filePath.getFileSystem(configuration)
+      if(fs.exists(filePath)) {
+        val cs = fs.getContentSummary(filePath)
```

Comment:
This kind of listing can potentially be expensive, and looks `getContentSummary` is recursive too. I think we should at least guard by a legacy configuration with updating the migration guide at https://github.com/apache/spark/blob/master/docs/sql-migration-guide.md

Reply:
As fs.getContentSummary() is a costly operation compared to fs.listStatus(), now I used fs.listStatus(...).listStatus !=0
Also added new legacy configuration settings and updated the SQL migration guide